### PR TITLE
[WIP] Add documentation outlining machine readable output

### PR DESCRIPTION
**Pull request by cdrage**
Adds a document that outlines all possible examples / options for each
command available.

Closes https://github.com/redhat-developer/odo/issues/1464

### CODE REVIEW

**Comment by cdrage** [comment]
Note, this is a WIP as I'm probably missing a lot more commands, although some are being deprecated, for example `odo app create` since we're removing app, as well as `odo app list`..

File: docs/machine-output.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
+# Machine Readable Output
+
+This document outlines all the machine readable output options and examples.
+
+`$ odo url create -o json | jq`
```
Comment (mik-dass) [inline docs/machine-output.md]: I think we should mention before that the `jq` tool is required for pretty formatting

Comment (surajnarwade) [reply docs/machine-output.md]: +1

Comment (cdrage) [reply docs/machine-output.md]: Done :+1:

**Comment by cdrage** [comment]
I'm going to close this for now :+1: will revisit this in the future.

**Pull request closed without merging**